src/rules/fileNameCasingRule.ts

```diff
@@ -92,15 +140,53 @@ export class Rule extends Lint.Rules.AbstractRule {
         }
     }
 
+    private static isValidRegExp(regExpString: string) {
+        try {
+            RegExp(regExpString);
+            return true;
+        } catch (e) {
+            return false;
+        }
+    }
+
+    private static findApplicableCasing(
+        fileBaseName: string,
+        fileNameCasings: FileNameCasings
+    ): Casing | null {
```

Comment:
> `| null`

Why so many `null`s instead of `undefined`? It's generally preferable in JavaScript/TypeScript to prefer using `undefined`, as it's the default value. TSLint typically goes by that preference.

Unless you have a real need for using `null`, please switch to `undefined` as defaults.

Reply:
My bad, in my team I'm still fighting for `strictNullChecks`, so as of now we don't really explicitly work with either `undefined` or `null`. But we're getting there 😊 While changing the code to use `undefined` I realized that's way more intuitive, thanks for the suggestion!